src/cli_plugin/install/pack.js

```diff
@@ -71,6 +71,15 @@ async function mergePackageData(settings, packages) {
     pkg.name = _.get(packageInfo, 'name');
     pkg.path =  resolve(settings.pluginDir, pkg.name);
 
+    // Plugins must specify their version, and by default that version should match
+    // the version of kibana down to the patch level. If these two versions need
+    // to diverge, they can specify a kibana.version to indicate the version of
+    // kibana the plugin is intended to work with.
+    pkg.kibanaVersion = _.get(packageInfo, 'kibana.version');
```

Comment:
You can use `_.get(packageInfo, 'kibana.version', pkg.version);` and get rid of the check below.